src/webots/nodes/WbHingeJoint.cpp

```diff
@@ -348,9 +348,9 @@ void WbHingeJoint::postPhysicsStep() {
 void WbHingeJoint::updatePosition() {
   // Update triggered by an artificial move, i.e. a move caused by the user or a Supervisor
   const WbJointParameters *const p = parameters();
-  assert(p);
+
   if (solidReference() == NULL && solidEndPoint())
-    updatePosition(p->position());
+    updatePosition(p ? p->position() : 0.0);
```

Comment:
Actually, balljoint and hinge2joint already implement fallbacks in case no parameter is present:

- https://github.com/cyberbotics/webots/blob/revision/src/webots/nodes/WbBallJoint.cpp#L228
- https://github.com/cyberbotics/webots/blob/revision/src/webots/nodes/WbHinge2Joint.cpp#L533

So I think this should be fixed only in WbHingeJoint and WbSliderJoint.

However, I remark now that the default value is mPosition, and not 0.0 in these cases. I have to check what is true there...

Reply:
And what about the assert :wink:

Reply:
These asserts are wrong, this case can obviously occur. I'm dropping them, thank you.

Reply:
I checked, and setting an default value to `mPosition` seems more logical and to work smoothly (I checked in the problematic test suite tests, and in the worlds I tested to fix this).